### PR TITLE
ClusterRegistry diagnostic: fix address mismatch

### PR DESCRIPTION
**Pull request by sosiouxme**
The registry for imagestreams was previously recorded as an IP, which
could change if the registry service were re-created.  Now it is a
cluster hostname, which should be unchanging even if re-created. Just
ensure it is the right hostname.

fix bug 1488059
https://bugzilla.redhat.com/show_bug.cgi?id=1488059

### CODE REVIEW

File: pkg/diagnostics/cluster/registry.go [hunk 1]
```diff
@@ -338,8 +338,11 @@ func (d *ClusterRegistry) verifyRegistryImageStream(service *kapi.Service, r typ
...
 	cacheHost := strings.SplitN(imgStream.Status.DockerImageRepository, "/", 2)[0]
-	serviceHost := fmt.Sprintf("%s:%d", service.Spec.ClusterIP, service.Spec.Ports[0].Port)
-	if cacheHost != serviceHost {
-		r.Error("DClu1019", nil, fmt.Sprintf(clRegISMismatch, registryName, serviceHost, cacheHost))
+	// the registry for imagestreams was previously recorded as an IP, which could change if the registry service were re-created.
+	// Now it is a cluster hostname, which should be unchanging even if re-created. Just ensure it is the right hostname.
+	serviceIpPort := fmt.Sprintf("%s:%d", service.Spec.ClusterIP, service.Spec.Ports[0].Port)
+	serviceHostPort := fmt.Sprintf("%s.%s.svc:%d", registryName, service.ObjectMeta.Namespace, service.Spec.Ports[0].Port)
+	if cacheHost != serviceIpPort && cacheHost != serviceHostPort {
+		r.Error("DClu1019", nil, fmt.Sprintf(clRegISMismatch, registryName, serviceIpPort, cacheHost))
```
Comment (liggitt) [inline pkg/diagnostics/cluster/registry.go]: Should the message say something about serviceHostPort?

Comment (sosiouxme) [reply pkg/diagnostics/cluster/registry.go]: I guess that would make sense, thanks.

**Comment by sosiouxme** [comment]
@liggitt addressed that.

Is there a magic bot command to indicate this should be cherry-picked to 3.6?

**Pull request merged**